pkg/auth/disabled_authhandler.go

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package auth
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/auth/certs"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+// disabledAuthHandler implements an authHandler for auth type disable by just authenticate every request.
+type disabledAuthHandler struct {
```

Comment:
there isn't an authhandler for type `disabled` -> these requests should never be passed to the auth manager (and therefore an auth handler)

-> instead of renaming from `nullAuthHandler` to `disabledAuthHandler` -> i would recommend to rename this one to `alwaysPassAuthHandler` to be aligned with the `always-fail` authhandler.

please ntoe that in comparison to the "always-fail", the "always-pass" shouldn't be available via configuration and only be used for unit-testing